### PR TITLE
Update Makefile for Multi-Arch build

### PR DESCRIPTION
**Pull request by fdelucchijr**
Using  [vinodkumark369](https://github.com/vinodkumark369) script. Tested in WSL.

### CODE REVIEW

**Comment by vinodkumark369** [comment]
@fdelucchijr It is not my script, that is already present in the github action for release of bin image. That is why I am confused about how the image is actually built. I am assuming it is built using circle-ci.

**Comment by fdelucchijr** [comment]
> That is already present in the github action for release of bin image.

🤔, I think the CD script is .circleci/config.yml, in the last step triggers the makefile I'd change in this PR.

**Comment by vinodkumark369** [comment]
@fdelucchijr I think this should be enough to build a multi-arch image. My only concern will be the image `okteto/golang-ci:1.16` used in the release job. That image is also build for `linux/amd64` only. I don't know what it is supposed to do, so maybe @pchico83 could check that.

**Comment by vinodkumark369** [comment]
@fdelucchijr @pchico83 Do we really need the Makefile anymore? Could that just be a step inside the release job?

**Comment by pchico83** [comment]
> @fdelucchijr I think this should be enough to build a multi-arch image. My only concern will be the image `okteto/golang-ci:1.16` used in the release job. That image is also build for `linux/amd64` only. I don't know what it is supposed to do, so maybe @pchico83 could check that.

Since that image is just running the `docker build` command, I think it's fine if it's only amd64

**Comment by pchico83** [comment]
> @fdelucchijr @pchico83 Do we really need the Makefile anymore? Could that just be a step inside the release job?

Yeap, +1 for removing `Makefile`

**Comment by pchico83** [comment]
@fdelucchijr you need to update https://github.com/okteto/bin-image/blob/master/.circleci/config.yml#L13 to the tag 1.17.11. We have added the `buildx` plugin to the image

**Comment by pchico83** [comment]
> @fdelucchijr you need to update https://github.com/okteto/bin-image/blob/master/.circleci/config.yml#L13 to the tag 1.17.11. We have added the `buildx` plugin to the image

@fdelucchijr done already here https://github.com/okteto/bin-image/pull/5

**Comment by fdelucchijr** [comment]
Hi @pchico83! As I see in Okteto/devenv#96 your trying to build the images in GH actions, I think that’s a wonderful idea and have a lot more experience to contribute. I’ve removed the CircleCI pipeline and added an almost identical GH Actions pipeline we use at our business. 
It pushes the image to the repo registry and to Docker Hub, because of that it needs to get Docker credential from repo secrets. Hope it helps and centralize a little the build process with the repo.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Docker Image CI
+
+on:
+  pull_request:
+    branches: [master]
```
Comment (pchico83) [inline .github/workflows/build.yaml:5]: we should only build on PRs, and push on merges to `master`

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Pushing Image to Github Container Registry
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
```
Comment (pchico83) [inline .github/workflows/build.yaml]: this should only be done on merges to `master`, not on PRs

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          username: "${{ secrets.DOCKER_USERNAME }}"
```
Comment (pchico83) [inline .github/workflows/deploy.yaml]:
```suggestion
          username: "${{ secrets.DOCKER_USER }}"
```

File: .github/workflows/deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          password: "${{ secrets.DOCKER_TOKEN }}"
```
Comment (pchico83) [inline .github/workflows/deploy.yaml]:
```suggestion
          password: "${{ secrets.DOCKER_PASS }}"
```

File: .github/workflows/deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
```
Comment (pchico83) [inline .github/workflows/deploy.yaml]: is there any reason to use `v1` here and `v2` here?
https://github.com/okteto/bin-image/pull/4/files#diff-4f9e38227ed64fefb17f4668a7ac4ab55b6149994d5ac2fd96182d5958479b54R56

File: .github/workflows/deploy.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          username: "${{ secrets.DOCKER_USERNAME }}"
+          password: "${{ secrets.DOCKER_TOKEN }}"
+
+      - name: Login to GitHub Container Registry
```
Comment (pchico83) [inline .github/workflows/deploy.yaml]: Let's just push to Dockerhub. One reason to use GH container registry is unlimited pulls, but we are verified publishers in Dockerhub and our images have unlimited pulls for everyone for free

File: .github/workflows/deploy.yaml [hunk 7]
```diff
@@ -0,0 +1,81 @@
+name: Docker Image CD
+
+on:
+  workflow_dispatch:
+    inputs:
+      bin_tag:
```
Comment (pchico83) [inline .github/workflows/deploy.yaml]: This is only executed when there is a new tag, right?

File: .github/workflows/deploy.yaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+      - name: Pushing Image to Github Container Registry
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          build-args: |
+            OKTETO_BIN_TAG=${{ github.event.inputs.bin_tag }}
+          tags: |
+            okteto/bin:latest
```
Comment (pchico83) [inline .github/workflows/deploy.yaml]: let's remove `latest` here. the tag could be for an older commit than the current master branch

**Review by pchico83 — COMMENTED** [review]
@fdelucchijr I love this one! I left some minor comments. Once this is merged, I will reuse your actions for https://github.com/okteto/devenv. Thanks for taking care of this!

File: .github/workflows/deploy.yaml (hunk 7 above)
Comment (fdelucchijr) [reply .github/workflows/deploy.yaml]: workflow_dispatch is a manual trigger, I don't know where can I receive the state of the binaries to dispatch the pipeline 🤔

File: .github/workflows/deploy.yaml (hunk 6 above)
Comment (fdelucchijr) [reply .github/workflows/deploy.yaml]: Perfect! I didn't know  about that feature in Dockerhub 👍🏻

File: .github/workflows/deploy.yaml (hunk 7 above)
Comment (pchico83) [reply .github/workflows/deploy.yaml]: On other repos we do this:
```
on:
  release:
    types:
      - published
``` 
and then we get the value of the tag like this:
```
    - id: tag
      name: Get the tag
      run: echo ::set-output name=TAG::${GITHUB_REF#refs/tags/}

    - name: "Push"
      run: |
        export TAG=${{ steps.tag.outputs.TAG }}
        make push
```

Comment (fdelucchijr) [reply .github/workflows/deploy.yaml]: Ohhh! Ok!, I didn't notice your uploading the bin as a release in the repo. That's perfect, I will add the same trigger and removing the workflow_dispatch.

File: .github/workflows/deploy.yaml [hunk 9]
```diff
@@ -61,9 +55,9 @@ jobs:
           context: .
           push: true
           build-args: |
-            OKTETO_BIN_TAG=${{ github.event.inputs.bin_tag }}
+            OKTETO_BIN_TAG=${{ github.event.release.tag_name }}
           tags: |
-            okteto/bin:${{ github.event.inputs.bin_tag }}
+            okteto/bin:${{ github.event.release.tag_name }}
           platforms: linux/amd64,linux/arm64
```
Comment (fdelucchijr) [inline .github/workflows/deploy.yaml:61]: This should do the trick based in https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#release

**Review by pchico83 — APPROVED** [review]
Thanks a lot @fdelucchijr!
This was a lot of fun, I am going to use these actions in many places now :-)

**Pull request merged**